QMLComponents/controls/jaspcontrol.cpp

```diff
@@ -633,9 +674,22 @@ void JASPControl::rScriptDoneHandler(const QString &)
 	throw std::runtime_error("runRScript done but handler not implemented!\nImplement an override for RScriptDoneHandler\n");
 }
 
-
 void JASPControl::_setFocus()
 {
 	if (!hasActiveFocus())
 		setFocus(false);
 }
+
+void JASPControl::_notifyFormOfActiveFocus()
+{
+	if (_form)
+	{
+		if (!hasActiveFocus())
+		{
+			_form->setActiveJASPControl(nullptr);
+			_activeJASPControl = false;
+		}
+		if(_activeJASPControl)
+			_form->setActiveJASPControl(this);
+	}
+}
```

Comment:
I see a concurrency problem here: If the new active JASP control runs first this _notifyFormOfActiveFocus method, and then the JASP control that loses the active focus, then the form will lose the active JASP control.
I see that you have solved this problem by adding the line:
`if(!control && _activeJASPControl && _activeJASPControl->hasActiveFocus())` in setActiveJASPControl...
What I'd rather do is:
```
setActiveJASPControl(JASPControl* control, hasActiveFocus)
{
  bool emitSignal = false;
  if (hasActiveFocus)
  {
     if (_activeJASPControl != control) emitSignal = true;
     _activeJASPControl = control
  }
  else if (control == _activeJASPControl)
  {
     if (_activeJASPControl) emitSignal = true;
     _activeJASPControl = nullptr;
  }

  if (emitSignal)
     emit activeJASPControlChanged();
}
```
Also I would rename the JASPControl `_activeJASPControl` to `_hasActiveFocus`  because it is quite confusing with the form `_activeJASPControl`